src/XperienceCommunity.HCaptcha/Components/FormComponents/HCaptchaFormComponent.cshtml

```diff
@@ -22,9 +22,35 @@
                 'size': '@Model.Properties.Layout',
             });
         };
+
+        // Add function to load array in case multiple hcaptcha
+        window.hcaptchatoload = window.hcaptchatoload || [];
+
+        // if not loaded, then start loading and add current initializer to queue.
+        if(!window.hcpatchaloading) {
+            window.hcpatchaloading = true;
+            window.hcaptchaloaded = false;
+
+            window.hcaptchatoload.push(@(Html.Raw(renderFunctionName)));
+
+            // Load script only once and call
+            var script = document.createElement("script");  // create a script DOM node
+            script.src = "https://js.hcaptcha.com/1/api.js?render=explicit&hl=@Model.Language";  // set its src to the provided URL
+            document.head.appendChild(script);
+            script.addEventListener('load', function(e) {
+                 window.hcaptchaloaded = true;
+                 for(var i=0; i< window.hcaptchatoload.length; i++) {
+                      window.hcaptchatoload[i]();
+                 }
+            })
+        } else if(window.hcaptchaloaded) {
+            // if already loaded, then run function
+            @(Html.Raw(renderFunctionName))();
+        } else {
+            // If both loading and not already loaded, add to queue to run once loaded.
+            window.hcaptchatoload.push(@(Html.Raw(renderFunctionName)));
+        }
     </script>
-    <script src="https://js.hcaptcha.com/1/api.js?onload=@Html.Raw(renderFunctionName)&render=explicit&hl=@Model.Language"
-    async defer></script>
```

Comment:
Since we are manually loading the script now, we don't need the unique function name.

Want to change the entire script to this (which simplifies the function calling and loading state management)?

```suggestion
<script>
    (function () {
        'use strict';

        function renderHCaptcha() {
            window.hcaptcha.render('@hcaptchaDivId', {
                'sitekey': '@Model.PublicKey',
                'theme': '@Model.Properties.Theme',
                'size': '@Model.Properties.Layout',
            });
        };

        if (window.hcaptchaScriptStatus === 'LOADED') {
            renderHCaptcha();

            return;
        }

        window.hcaptchatoload = window.hcaptchatoload || [];

        if (window.hcaptchaScriptStatus === 'LOADING') {
            window.hcaptchatoload.push(renderHCaptcha);

            return;
        }

        if (window.hcaptchaScriptStatus !== 'LOADING' && window.hcaptchaScriptStatus !== 'LOADED') {
            window.hcaptchaScriptStatus = 'LOADING';
            window.hcaptchatoload.push(renderHCaptcha);

            var script = document.createElement("script");
            script.src = "https://js.hcaptcha.com/1/api.js?render=explicit&hl=@Model.Language";
            script.addEventListener('load', function (e) {
                window.hcaptchaScriptStatus = 'LOADED';

                for (var i = 0; i < window.hcaptchatoload.length; i++) {
                    window.hcaptchatoload[i]();
                }
            });
            
            document.head.appendChild(script);

            return;
        }
    })();
</script>
```